src/GovUk.Education.ExploreEducationStatistics.Common/Cancellation/AddTimeoutCancellationAttribute.cs

```diff
@@ -0,0 +1,68 @@
+#nullable enable
+using System;
+using AspectInjector.Broker;
+using Microsoft.Extensions.Configuration;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Cancellation
+{
+    /// <summary>
+    /// An attribute that, when used upon a method that accepts a
+    /// CancellationToken as its final parameter, provides a 
+    /// CancellationToken with the given timeout in milliseconds
+    /// to the method as its CancellationToken argument.
```

Comment:
Just wondering if there's any way with the 'Usage' argument of attributes that it could be constrained as an attribute of the `CancellationToken` rather than adding at a method level? I suppose at the moment it could be added to a method without a `CancellationToken` and it would not be triggered.

Reply:
Couldn't find it, but see answer to Nick's similar comment.